src/topic.rs

```diff
@@ -0,0 +1,21 @@
+/// A message queue topic name to which messages can be published
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
+pub struct Topic(pub &'static str);
```

Comment:
I think it should be a `Cow<'static, str>` while we're at it. Topic choice could be dynamic

Reply:
IIRC maru was against the idea of dynamic topics, but I'll let them to chime in.

Reply:
I made the field private which should make it possible to extend this to `Cow` in a backwards compatible manner in the future.